### ISSUE TITLE
Pullup as a reddit style layer for pull requests and issues.

### ISSUE THREAD

**Issue by howardroark** [larvalabs/pullup#382]
What if pullup was a tool that anyone could connect to a github repository.  From there it could offer a reddit/hacker news voting layer for pull requests and issues.  It could be built around GitHub accounts and use the API.  I think it would be a great way to understand what users are looking for an cut out a lot of noise.

**Comment by megamattron** [comment]
That would be an interesting direction, basically auto list any PR or issue and let people vote. It'd be a lot different than what we have here, but could be a good fork.

**Comment by rickhanlonii** [comment]
I like this as an addition to what we already have. 

I imagine we could have different categories of posts (Designer News style) and you can view only certain categories at a time (subreddit style). So the front page would show the highest voted article/news/repo/pull, but you can view and participate in the sub-communities, so to speak.

Once that core is there, then you could essentially allow filtering by repo for the code/repo/issue category.

That would take a few steps:

1. Implement categories on the backend
2. Add categories indicators on the front-end
3. Add categorization to adding a post (likely automatically at first)
4. Add routes for viewing by a particular category
5. Add handling for filtering by repo

**Comment by howardroark** [comment]
Yeah really anything can be in the "stacking order"! Just need a base schema and a uniquely extended schemas for each "thing".  Then you can filter as you like.  Just need a reusable UI +/- mechanism and a tool that converts GitHub data to the schemas and stores them.

**Comment by rickhanlonii** [comment]
We could create a /issues for any arbitrary public repo:

- pullup.io/larvalabs/pullup/issues
- pullup.io/reactjs/redux/issues

**Comment by howardroark** [comment]
That would be great!  Just have a basic algorithm that allows any GitHub id to vote.

**Comment by howardroark** [comment]
https://gitsup.com/
https://gitsup.com/larvalabs/pullup/issues

Was a great excuse to learn meteor.js :)  Still in a beta state, but does the basic job.

**Comment by rickhanlonii** [comment]
That's really neat @howardroark! 

Are you interested in integrating that into pullup directly? I'm happy to help in any way I can.

**Comment by howardroark** [comment]
Hey @rickhanlonii I thought about how this could be done for while, but couldn't quite see how it would make cohesive sense aside from changing what the project is about.  I think this project is more of an effort to build a community than anything else.  

With that said... I think there is an opportunity to distill a UI toolkit and API schemas for content voting systems.  All this app really does is talk to the github api via the browser and store votes as a schema and also update another schema that offers a tally of all votes related to a repo.  Other things could source that data... like if someone wanted to embed a list in their project site and feed from the core data.

Too bad the "pullup" org handle is taken so many projects couldn't be under the same umbrella.
